SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -513,6 +513,16 @@ void ComponentReader::ApplyArray(Schema_Object* Object, Schema_FieldId FieldId,
 		ApplyProperty(Object, FieldId, *ArrayObjectReferences, i, Property->Inner, ArrayHelper.GetRawPtr(i), ElementOffset, ElementOffset, ParentIndex, bOutReferencesChanged);
 	}
 
+	const int32 MaxOffset = Count * Property->Inner->ElementSize;
+
+	for (auto It = ArrayObjectReferences->CreateIterator(); It; ++It)
```

Comment:
As we're applying all objects in the step just prior to this, couldn't we just empty ArrayObjectReferences prior to applying any data and rely on any existing mappings to get remapped?

Reply:
Yep, that should work.  I had a misunderstanding about how ApplyProperty worked internally with object references.  Will update.

Reply:
The change to use Empty() has now been tested on NWX.